kubernetes/resource_kubernetes_secret_v1_data.go

```diff
@@ -0,0 +1,227 @@
+			},
+		},
+	}
+}
+
+func resourceKubernetesSecretV1DataCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	// Sets the resource id based on the metadata
+	d.SetId(buildId(metadata))
+
+	//Calling the update function ensuring resource config is correct
+	diag := resourceKubernetesSecretV1DataUpdate(ctx, d, m)
+	if diag.HasError() {
+		d.SetId("")
+	}
+	return nil
+}
+
+// Retrieves the current state of the k8s secret, and update the current sate
+func resourceKubernetesSecretV1DataRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	conn, err := m.(KubeClientsets).MainClientset()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	//extracting ns and name from res id
+	namespace, name, err := idParts(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	// Retrieve the K8s secret
+	secret, err := conn.CoreV1().Secrets(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Handle case where the Secret is not found
+			return diag.Diagnostics{{
+				Severity: diag.Warning,
+				Summary:  "Secret deleted",
+				Detail:   fmt.Sprintf("The underlying secret %q has been deleted. You should recreate the underlying secret, or remove it from your configuration.", name),
+			}}
+		}
+		return diag.FromErr(err)
+	}
+
+	// Extract managed data from the secret
+	managedData, err := getManagedSecretData(secret)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// filter out the data not managed by terraform
+	configuredData := d.Get("data").(map[string]interface{})
+	for k := range managedData {
+		if _, exists := configuredData[k]; !exists {
+			delete(managedData, k)
+		}
+	}
+	// Update the state with the managed data
+	d.Set("data", managedData)
+	return nil
+}
+
+// extracts data from the secret that is managed by terraform
+func getManagedSecretData(secret *v1.Secret) (map[string]interface{}, error) {
+	managedData := make(map[string]interface{})
+
+	//looping through all data in the secret
+	for key, value := range secret.Data {
+		// decode base64-encoded value
+		decodedValue, err := base64.StdEncoding.DecodeString(string(value))
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode value for key %q: %w", key, err)
+		}
+
+		// just storing the decoded value I got in the managed data map
+		managedData[key] = string(decodedValue)
+	}
+	return managedData, nil
+
+}
+
+func resourceKubernetesSecretV1DataUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	conn, err := m.(KubeClientsets).MainClientset()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	name := metadata.GetName()
+	namespace := metadata.GetNamespace()
+
+	_, err = conn.CoreV1().Secrets(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
```

Comment:
After running this on my machine the `inconsistent apply` is coming from this block not being entered. We should be expecting a `diag.Errorf("The Secret %q does not exist", name)` if no secret has been applied yet it doesn't do that. 
```
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to kubernetes_secret_v1_data.test, provider
│ "provider[\"registry.terraform.io/hashicorp/kubernetes\"]" produced an unexpected new value: Root
│ object was present, but now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

My first glance from it is that it does successfully perform a GET request but the resp tells us nothing is found.

```
ransport.go:160 new_logger_warning="This log was generated by a subsystem logger that wasn't created before being used. Use tflog.NewSubsystem to create this logger before it is used." tf_http_res_status_code=404 @module=kubernetes.Kubernetes X-Kubernetes-Pf-Flowschema-Uid=d717b834-9d7f-4086-8519-e6b121115470 tf_http_trans_id=fb8f7e58-365d-d708-9c1b-07d7cb9dc739
  tf_http_res_body=
  | {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"secrets \"jaylon-pr-test\" not found","reason":"NotFound","details":{"name":"jaylon-pr-test","kind":"secrets"},"code":404}
```

This is a different behavior to what's in `config_map_data` since a get on a resource that doesn't exist would return an `err`.

Reply:
I am no longer getting this error, it was an annoying bug I had haha!